pkg/inngest/function.go

```diff
@@ -109,6 +110,37 @@ type Throttle struct {
 	Key *string `json:"key,omitempty"`
 }
 
+func (t *Throttle) UnmarshalJSON(in []byte) error {
+	if t == nil {
+		t = &Throttle{}
+	}
+
+	var err error
+	input := struct {
+		Limit  uint    `json:"limit"`
+		Period string  `json:"period"`
+		Burst  uint    `json:"burst"`
+		Key    *string `json:"key,omitempty"`
+	}{}
+	if err = json.Unmarshal(in, &input); err != nil {
+		return err
+	}
+
+	t.Limit = input.Limit
+	t.Burst = input.Burst
+	t.Key = input.Key
+	t.Period, err = str2duration.ParseDuration(input.Period)
+
+	// Normalization
+	if t.Limit == 0 {
+		t.Limit = 1
+	}
+	if t.Burst == 0 {
+		t.Burst = 1
+	}
```

Comment:
Can we catch negative values here too?
```suggestion
	// Normalization
	if t.Limit <= 0 {
		t.Limit = 1
	}
	if t.Burst <= 0 {
		t.Burst = 1
	}
```

Reply:
Should negative value handling be in validation or normalization?

Reply:
These are uints, so it's already handled.